interface/patient_file/printed_fee_sheet.php

```diff
@@ -361,20 +360,40 @@ function printlog_before_print() {
 
 // Loop on array of PIDS
 $saved_pages = $pages; //Save calculated page count of a single fee sheet
+$loop_idx = 0; // counter for appt list
+
 
 foreach ($pid_list as $pid) {
+
+    $apptdate = $apptdate_list[$loop_idx]; // parallel array to pid_list
+    $appointment = fetchAppointments( $apptdate, $apptdate, $patient_id=$pid );  // Only expecting one row for pid
+
+    // Set Pagebreak for multi forms
+    if ($form_fill == 2) {
+        $html .= "<div class=pagebreak>\n";
+    } else {
+        $html .= "<div>\n";
+    }
+
     if ($form_fill) {
         // Get the patient's name and chart number.
         $patdata = getPatientData($pid);
+        // Get the referring providers info
+        $referDoc = getUserIDInfo($patdata['ref_providerID']);
     }
 
 // This tracks our position in the $SBCODES array.
     $cindex = 0;
 
     while (--$pages >= 0) {
         $html .= genFacilityTitle(xl('Superbill/Fee Sheet'), -1, $logo);
-
-        $html .="
+        $html .= '<table style="width: 100%"><tr>' .
+            '<td>Patient: <span style="font-weight: bold;">' . $patdata['fname'] . ' ' . $patdata['mname'] . ' ' . $patdata['lname'] . '</span></td>' .
+            '<td>DOB: <span style="font-weight: bold;">' . date("m-d-Y", $patdata['DOB']) . '</span></td>' .
+            '<td>Date of Service: <span style="font-weight: bold;">' . date("m-d-Y", $appointment['pc_eventDate']) . '</span></td>' .
+            '<td>Ref Prov: <span style="font-weight: bold;">' . $referDoc['fname'] . ' ' . $referDoc['lname'] . '</span></td>' .
```

Comment:
need translations and escaping:
```
            '<td>' . xlt('Patient') . ': <span style="font-weight: bold;">' . text($patdata['fname']) . ' ' . text($patdata['mname']) . ' ' . text($patdata['lname']) . '</span></td>' .
            '<td>' . xlt('DOB') . ': <span style="font-weight: bold;">' . text(date("m-d-Y", $patdata['DOB'])) . '</span></td>' .
            '<td>' . xlt('Date of Service') . ': <span style="font-weight: bold;">' . text(date("m-d-Y", $appointment['pc_eventDate'])) . '</span></td>' .
            '<td>' . xlt('Ref Prov') . ': <span style="font-weight: bold;">' . text($referDoc['fname']) . ' ' . text($referDoc['lname']) . '</span></td>' .
```

Reply:
Thanks @bradymiller.  I can see you are trying to handle escaping for text in case it is required based on content.  I tried to find the xlt() and text() in both PHP itself and within related openemr *inc files.  Not sure where they are or exactly what / how they do it.  But I would agree to conform to your project's best practices.  I'm guessing @stephenwaite can use your code changes as is.  I also commented on your other comment above, but without the @ names (just in case you only get one notification).  If you can point me to where the xlt() and text() code is located, that would be helpful.  Thanks.

Reply:
hi @rpulcer6488, see [this url](https://github.com/openemr/openemr/blob/master/library/htmlspecialchars.inc.php#L72)